src/manifolds/SymmetricPositiveDefiniteFixedDeterminant.jl

```diff
@@ -0,0 +1,155 @@
+@doc raw"""
+    SymmetricPositiveDefiniteFixedDeterminant{N,D} <: AbstractDecoratorManifold{ℝ}
+
+The manifold of symmetric positive definite matrices of fixed determinant ``d > 0``, i.e.
+
+````math
+\mathcal P_d(n) =
+\bigl\{
+p ∈ ℝ^{n × n} \ \big|\ a^\mathrm{T}pa > 0 \text{ for all } a ∈ ℝ^{n}\backslash\{0\}
+  \text{ and } \det(p) = d
+\bigr\}.
+````
+This manifold is modelled as a submanifold of [`SymmetricPositiveDefinite`](@ref)`(n)`.
+
+The tangent space is modelled the same as for [`SymmetricPositiveDefinite`](@ref)`(n)`
+and consists of all symmetric matrices with zero trace
+```math
+    T_p\mathcal P_d(n) =
+    \bigl\{
+        X \in \mathbb R^{n×n} \big|\ X=X^\mathrm{T} \text{ and } \operatorname{tr}(p) = 0
+    \bigr\},
+```
+since for a constant determinant we require that `0 = D\det(p)[Z] = \det(p)\operatorname{tr}(p^{-1}Z)` for all tangent vectprs ``Z``.
+Additionally we store the tangent vectors as `X=p^{-1}Z`, i.e. symmetric matrices.
+
+# Constructor
+
+    SymmetricPositiveDefiniteFixedDeterminant(n, d=1.0)
+
+generates the manifold $\mathcal P_d(n) \subset \mathcal P(n)$ of determinant ``d``,
+which defaults to 1.
+"""
+struct SymmetricPositiveDefiniteFixedDeterminant{N,D} <: AbstractDecoratorManifold{ℝ} end
```

Comment:
```suggestion
struct SymmetricPositiveDefiniteFixedDeterminant{N,TD<:Real} <: AbstractDecoratorManifold{ℝ}
    d::TD
end
```
I think determinant should be just a field.

Reply:
Yes, you might be right, there is not much reason to ever dispatch on that.